.pre-commit-config.yaml

```diff
@@ -0,0 +1,87 @@
+# See https://pre-commit.com for more information
```

Comment:
`pre-commit` is smart enough to only run checks on files that changed. So, it can be pretty quick. But if the full set of checks get run then it can be pretty slow. I don't think it should ever be any slower than `ci/code_checks.sh` though so I'm inclined to not worry about this in that PR, but we could open an issue and try to improve it in future PRs.

Reply:
Sounds good, that's a very useful feature (that `code_checks.sh` did not have!)